homeassistant/components/remote/harmony.py

```diff
@@ -192,13 +192,19 @@ def __init__(self, name, host, port, activity, out_path, delay_secs):
         self.hass.bus.async_listen_once(EVENT_HOMEASSISTANT_STOP, shutdown)
 
         _LOGGER.debug("%s: Connecting", self._name)
-        try:
-            await self._client.connect()
-        except aioexc.TimeOut:
-            _LOGGER.error("%s: Connection timed-out", self._name)
-        else:
-            # Set initial state
-            self.new_activity(self._client.current_activity)
+        while True:
+            try:
+                if await self._client.connect():
+                    break
+            except aioexc.TimeOut:
```

Comment:
Wouldn't it be better to use PlatformNotReady?

Reply:
Two questions:
1. Is there an example you can point me to for PlatformNotReady?
2. If stating PlatformNotReady, will HASS then retry and what will it call then?

Thx. :-)

Reply:
Lots of examples in the code base.
PlatformNotReady will trigger setup of platform after a timer of 60 seconds I believe.